### PR TITLE
feat(iconics): fetch blog api

### PR DESCRIPTION
**Pull request by Aklile-Yilma**

### CODE REVIEW

**Comment by Henok-Matheas** [comment]
You haven't provided a reference image for the task you worked on

File: Web/starter_project_iconics/store/features/api/blog-api.ts
Comment (Henok-Matheas) [inline Web/starter_project_iconics/store/features/api/blog-api.ts:1]: Use the common backend_url for the base url.

Comment (Henok-Matheas) [reply Web/starter_project_iconics/store/features/api/blog-api.ts:1]: change the reducer path to make it more descriptive, in this case the reducerPath should be something like "blog"

File: Web/starter_project_iconics/pages/_app.tsx
Comment (Henok-Matheas) [inline Web/starter_project_iconics/pages/_app.tsx:1]: the store name should be a global one, you are wrapping the whole app using the blogApi store.

File: Web/starter_project_iconics/pages/blog/index.tsx [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
+import { useGetBlogsQuery } from '@/store/features/api/blog-api'
+import Link from 'next/link'
 import { FaPlus } from 'react-icons/fa'
-import data from '../../data/blog/blog.json'
 
-const blogs = () => {
-  const blogsData: Blog[] = data
+const Blogs = () => {
+  const { data: blogsData, isLoading } = useGetBlogsQuery()
+  console.log(blogsData)
```
Comment (zerihunMoges) [inline Web/starter_project_iconics/pages/blog/index.tsx:8]: it's important to ensure that any debugging statements, including console.log, are removed before committing the code.

File: Web/starter_project_iconics/store/features/api/blog-api.ts
Comment (Henok-Matheas) [inline Web/starter_project_iconics/store/features/api/blog-api.ts:1]: change the reducer path to make it more descriptive, in this case the reducerPath should be something like "blog"

Comment (Henok-Matheas) [inline Web/starter_project_iconics/store/features/api/blog-api.ts:1]: Use the common backend_url for the base url set up in the config folder.

File: Web/starter_project_iconics/pages/blog/index.tsx
Comment (Henok-Matheas) [inline Web/starter_project_iconics/pages/blog/index.tsx:1]: you can change the isLoading part to be more of a skeleton.

Comment (Henok-Matheas) [inline Web/starter_project_iconics/pages/blog/index.tsx:1]: the Blogs component should have it's type defined like React.FC or something.

Comment (Henok-Matheas) [inline Web/starter_project_iconics/pages/blog/index.tsx:1]: some of the spacing is inconsistent, so use prettier to format it, you can use the extension if you want.

Comment (Henok-Matheas) [inline Web/starter_project_iconics/pages/blog/index.tsx:1]: you aren't handling the case where the API returns an error.

**Review by Henok-Matheas — CHANGES_REQUESTED** [review]
change these things before merging.

File: Web/starter_project_iconics/pages/blog/index.tsx [hunk 2]
```diff
@@ -24,12 +29,14 @@ const blogs = () => {
       </div>
```
Comment (Henok-Matheas) [inline Web/starter_project_iconics/pages/blog/index.tsx:30]: some of the spacing is inconsistent, so use prettier to format it, you can use the extension if you want.

File: Web/starter_project_iconics/pages/blog/index.tsx [hunk 3]
```diff
@@ -1,10 +1,15 @@
 import BlogTile from '@/components/blog/BlogTile'
-import { Blog } from '@/types/blog/blog'
+import { useGetBlogsQuery } from '@/store/features/api/blog-api'
+import Link from 'next/link'
 import { FaPlus } from 'react-icons/fa'
-import data from '../../data/blog/blog.json'
 
-const blogs = () => {
-  const blogsData: Blog[] = data
+const Blogs = () => {
```
Comment (Henok-Matheas) [inline Web/starter_project_iconics/pages/blog/index.tsx:6]: the Blogs component should have it's type defined like React.FC or something.

File: Web/starter_project_iconics/store/features/api/blog-api.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { Blog } from '@/types/blog/blog'
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
+
+const API_BASE_URL = 'http://localhost:3500'
+
+export const blogApi = createApi({
+  reducerPath: 'api/',
+  baseQuery: fetchBaseQuery({ baseUrl: API_BASE_URL }),
```
Comment (Henok-Matheas) [inline Web/starter_project_iconics/store/features/api/blog-api.ts:8]: Use the common backend_url for the base url set up in the config folder.

File: Web/starter_project_iconics/store/features/api/blog-api.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  endpoints: (builder) => ({
+    getBlogs: builder.query<Blog[], void>({
+      query: () => 'blogs',
```
Comment (Henok-Matheas) [inline Web/starter_project_iconics/store/features/api/blog-api.ts:11]: change the reducer path to make it more descriptive, in this case the reducerPath should be something like "blog"

**Review by Henok-Matheas — CHANGES_REQUESTED** [review]
Please address these issues prior to merging.

**Review by zerihunMoges — DISMISSED** [review]

**Review by Henok-Matheas — APPROVED** [review]
LGTM.

**Review by ruthgdb — APPROVED** [review]

**Pull request merged**